Source/Scene/Cesium3DTileset.js

```diff
@@ -0,0 +1,1821 @@
+         *
+         * @type {Event}
+         * @default new Event()
+         *
+         * @example
+         * tileset.allTilesLoaded.addEventListener(function() {
+         *     console.log('All tiles are loaded');
+         * });
+         *
+         * @see Cesium3DTileset#tilesLoaded
+         */
+        this.allTilesLoaded = new Event();
+
+        /**
+         * The event fired to indicate that a tile's content was unloaded.
+         * <p>
+         * The unloaded {@link Cesium3DTile} is passed to the event listener.
+         * </p>
+         * <p>
+         * This event is fired immediately before the tile's content is unloaded while the frame is being
+         * rendered so that the event listener has access to the tile's content.  Do not create
+         * or modify Cesium entities or primitives during the event listener.
+         * </p>
+         *
+         * @type {Event}
+         * @default new Event()
+         *
+         * @example
+         * tileset.tileUnload.addEventListener(function(tile) {
+         *     console.log('A tile was unloaded from the cache.');
+         * });
+         *
+         * @see Cesium3DTileset#maximumMemoryUsage
+         * @see Cesium3DTileset#trimLoadedTiles
+         */
+        this.tileUnload = new Event();
+
+        /**
+         * This event fires once for each visible tile in a frame.  This can be used to manually
+         * style a tileset.
+         * <p>
+         * The visible {@link Cesium3DTile} is passed to the event listener.
+         * </p>
+         * <p>
+         * This event is fired during the tileset traversal while the frame is being rendered
+         * so that updates to the tile take effect in the same frame.  Do not create or modify
+         * Cesium entities or primitives during the event listener.
+         * </p>
+         *
+         * @type {Event}
+         * @default new Event()
+         *
+         * @example
+         * tileset.tileVisible.addEventListener(function(tile) {
+         *     if (tile.content instanceof Cesium.Batched3DModel3DTileContent) {
+         *         console.log('A Batched 3D Model tile is visible.');
+         *     }
+         * });
+         *
+         * @example
+         * // Apply a red style and then manually set random colors for every other feature when the tile becomes visible.
+         * tileset.style = new Cesium.Cesium3DTileStyle({
+         *     color : 'color("red")'
+         * });
+         * tileset.tileVisible.addEventListener(function(tile) {
+         *     var content = tile.content;
+         *     var featuresLength = content.featuresLength;
+         *     for (var i = 0; i < featuresLength; i+=2) {
+         *         content.getFeature(i).color = Cesium.Color.fromRandom();
+         *     }
+         * });
+         */
+        this.tileVisible = new Event();
+
+        /**
+         * Optimization option. Determines if level of detail skipping should be applied during the traversal
```

Comment:
As far as I can tell, none of the documentation ever describes example what level of detail skipping is, why you may or may not want it, and what the tradeoffs are.